vip-lockout.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Plugin Name: Lockouts and Warnings for VIP Go
+ * Description: Displays a warning or lockout users from wp-admin
+ * Author: Automattic
+ * Author URI: http://automattic.com/
+ */
+
+class VIP_Lockout {
+
+	const USER_SEEN_WARNING_KEY = 'seen_lockout_warning';
+
+	const USER_SEEN_WARNING_TIME_KEY = 'seen_lockout_warning_time';
+
+	/**
+	 * VIP_Lockout constructor.
+	 */
+	public function __construct() {
+		add_action( 'admin_notices', [ $this, 'add_admin_notice' ], 1 );
```

Comment:
Did some testing and noticed two things:

1) It would be good to only hook into these if `VIP_LOCKOUT_STATE` and `VIP_LOCKOUT_MESSAGE` are set.

2) When locked, we'll need to hook into `pre_site_option_site_admins` and return an empty array when the site is locked.

Reply:
> 1. It would be good to only hook into these if `VIP_LOCKOUT_STATE` and `VIP_LOCKOUT_MESSAGE` are set. No need to add the extra overhead to all pageloads.

Good idea 👍🏽

> 2. When locked, we'll need to hook into `pre_site_option_site_admins` and return an empty array when the site is locked.

I can't find any references for a `pre_site_option_` filter. Did you mean `pre_option_`?

Reply:
> I can't find any references for a `pre_site_option_` filter.

It's a dynamic filter (Googling it is not easy :))

https://developer.wordpress.org/reference/hooks/pre_site_option_option/